### ISSUE TITLE
2.5.2 does’t have PyTorch's version. And I added it.

### ISSUE THREAD

**Issue by StevenJokes** [d2l-ai/d2l-en#989]
https://discuss.d2l.ai/t/auto-differentiation/35

**Comment by mli** [comment]
@AnirudhDagar

**Comment by AnirudhDagar** [comment]
Can you please explain the doubt exactly? I have also replied to the thread in the forum.

**Comment by StevenJokes** [comment]
The **first** reply is my 2.5.2's pytorch code. I use "detach" and "clone" to simulate what “copy” do in MXNET.**But which one is best, "detach" or "clone"?**

The **second** reply is about "the meaning of **d / a**" in 2.5.4. I tried another "b".The return of "b.grad == (d / b)" is false rather than true.**why?**

The **third** reply is my answer to 2.5.7. I'm still working on how to import plot in 2.4.

**Comment by AnirudhDagar** [comment]
Answer to **first** question.
`tensor.detach()`  creates a tensor that shares the same storage with  `tensor`  that does not require grad.
But `tensor.clone()` will also give you original tensor's requires_grad attributes. It is basically an exact copy including the computation graph.

Use  `detach()`  to remove a tensor from computation graph and use  `clone`  to copy the tensor while still keeping the copy as a part of the computation graph it came from.

The  **second**  answer about "the meaning of  **d / a**  " in 2.5.4.
“a.grad == (d / a)” is true because if you see how d is calculate using f(a). It is basically scaling a by some constant factor k. And if you were to do differentiate such a function say function `d=k*a` with respect to a then you would get that k. Hence these are true and obviously it won't hold for b because d is a function of which scales a and not b.

Answer to **third** question.
As i suggested earlier and you probably did it, you can simply `pip install d2l` to import torch from d2l.
If you want to import the specific rename imtorch.py just add this to the start of your code before making the import. -> 
```python
import sys
sys.path.insert(0, "../")
```

I replied to the forums thread as well. If you want we can continue the discussion there. Let me know if these answers solve your issue. :slight_smile:  
Should I close this?

**Comment by StevenJokes** [comment]
please look my new reply.

**Comment by StevenJokes** [comment]
My new reply:
https://discuss.d2l.ai/t/auto-differentiation/35/9

**Comment by AnirudhDagar** [comment]
Replied to the questions.
